src/formats/PDB.cpp

```diff
@@ -640,76 +688,77 @@ static std::string to_pdb_index(int64_t value, uint64_t width) {
 }
 
 struct ResidueInformation {
-    std::string atom_hetatm{ "HETATM" };
-    std::string resname { "XXX" };
-    std::string resid { "  -1" };
-    std::string chainid { "X" };
-    std::string inscode { " " };
-    std::string comp_type { "" };
+    std::string atom_hetatm = "HETATM";
+    std::string resname;
+    std::string resid;
+    std::string chainid;
+    std::string insertion_code;
+    std::string composition_type;
+    std::string segment;
 };
 
-static ResidueInformation get_residue_strings(optional<const Residue&> residue_opt, int64_t& max_resid) {
-
-    ResidueInformation res_info;
+static ResidueInformation get_residue_information(optional<const Residue&> residue_opt, int64_t& max_resid) {
+    ResidueInformation info;
 
     if (!residue_opt) {
         auto value = max_resid++;
-        res_info.resid = to_pdb_index(value, 4);
-
-        return res_info;
+        info.resid = to_pdb_index(value, 4);
+        return info;
     }
 
     auto residue = *residue_opt;
 
-    res_info.resname = residue.name();
     if (residue.get<Property::BOOL>("is_standard_pdb").value_or(false)) {
         // only use ATOM if the residue is standardized
-        res_info.atom_hetatm = "ATOM  ";
+        info.atom_hetatm = "ATOM  ";
     }
 
-    if (res_info.resname.length() > 3) {
-        warning("PDB writer", "residue '{}' name is too long, it will be truncated", res_info.resname);
-        res_info.resname = res_info.resname.substr(0, 3);
+    info.resname = residue.name();
+    if (info.resname.length() > 3) {
+        warning("PDB writer", "residue '{}' name is too long, it will be truncated", info.resname);
+        info.resname = info.resname.substr(0, 3);
     }
 
-    if (residue.id()) {
-        res_info.resid = to_pdb_index(residue.id().value() - 1, 4);
+    // if there is a residue, and it has no resid, set the resid to -1
+    info.resid = to_pdb_index(residue.id().value_or(-1) - 1, 4);
```

Comment:
Since -1 is allowed as a valid residue ID, would it be a good idea to change this default?

Reply:
I agree! The default will only be used if the frame contains residues with name but no id. Do you have a better idea for the default? We could use the largest/smallest possible id; or 0 which does not seems to be used a lot as resid.

Reply:
Actually, we don't have to invent the residue id if it is missing: the most probable reason why it is missing in the first place is that we read another PDB file without residue id. I'll change this too =)